### PR TITLE
August Update - YAML config, CLI arg, HTML class attributes

### PR DESCRIPTION
**Pull request by theyorubayesian**
I'm creating an update to this tool because it is being used in the AI Saturdays Cohort 7 course materials. 

Here are a list of changes:

- HTML class attributes and website category links have been moved to `env.yml`. This will make them easier to update in future.
- I added a `spread` command line argument for when the tools is used to collect a finite number of articles. Without this argument, the tool traverses the entire website collecting links before writing the specified number of articles. `spread` ensures that only the required number of links are collected per category. 
- Updated the README to reflect these changes.

### CODE REVIEW

**Comment by keleog** [comment]
thanks for the PR

File: scraper.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def get_page_soup(url:str) -> BeautifulSoup:
     return page_soup
 
 
-def get_urls(category_url:str, category:str, time_delay:bool) -> list:
+def get_urls(category_url:str, category:str, time_delay:bool, articles_per_category: int = None) -> list:
```
Comment (keleog) [inline scraper.py]: pls import `Optional` from `typing` and modify this to show that the default is `None`. 

So it becomes `articles_per_category: Optional[int] = None`

File: scraper.py [hunk 2]
```diff
@@ -235,16 +243,25 @@ def scrape(output_file_name:str, no_of_articles:int, category_urls:dict, time_de
 
     # specify categories to scrape
     if params.categories != "all":
-        categories = params.categories.split(",")
-        categories = {category:ALL_CATEGORIES[category] for category in categories}
+        categories = params.categories.upper().split(",")
```
Comment (keleog) [inline scraper.py:253]: why is this necessary?

File: scraper.py [hunk 3]
```diff
@@ -235,16 +243,25 @@ def scrape(output_file_name:str, no_of_articles:int, category_urls:dict, time_de
...
+        categories = {category: ALL_CATEGORIES[category] for category in categories}
     else:
         categories = ALL_CATEGORIES
 
+    articles_per_category = None
+    if params.no_of_articles > 0 and params.spread:
+        if "most_popular" in categories:
+            # most_popular only has ne page
```
Comment (keleog) [inline scraper.py]: `one page`

File: scraper.py [hunk 4]
```diff
@@ -235,16 +243,25 @@ def scrape(output_file_name:str, no_of_articles:int, category_urls:dict, time_de
...
+            articles_per_category = (params.no_of_articles-10) // (len(categories)-1)
```
Comment (keleog) [inline scraper.py:263]: can you leave a comment that we subtract 10 because there are 10 most popular articles?

File: README.md [hunk 5]
```diff
@@ -24,12 +24,20 @@ pip install beautifulsoup4
 
 ```--time_delay``` : If set to True, there will be a delay of 10 seconds between consecutive url requests (recommended). Default = ```True```
 
+```--spread```: If passed, number of articles collected is spread across all categories passed in. If `most_popular` in categories, all its articles are collected and the remainder is spread across other categories.
```
Comment (keleog) [inline README.md:27]: can you pls add this description, especially the part about the `most popular` category, to spread's argparse description in scraper.py

File: scraper.py [hunk 6]
```diff
@@ -190,7 +198,7 @@ def scrape(output_file_name:str, no_of_articles:int, category_urls:dict, time_de
 
     with open(output_file_name, "w") as csv_file:
         headers = ["headline", "text", "category", "url"]
-        writer = csv.DictWriter(csv_file, delimiter=",", fieldnames = headers)
+        writer = csv.DictWriter(csv_file, delimiter="\t", fieldnames = headers)
```
Comment (keleog) [inline scraper.py:208]: any reason for the `tsv` preference?

**Comment by keleog** [comment]
please rebase on the master branch, i made the types more explicit.

File: env.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+ARTICLE_COUNT_SPAN: lx-pagination__page-number qa-pagination-total-page-number
```
Comment (keleog) [inline env.yml]: can you rename `env.yml` to `config.yml`? i think config is a better descriptor.

File: scraper.py (hunk 2 above)
Comment (theyorubayesian) [reply scraper.py:253]: The YAML config uses upper case for its keys but categories are likely to be passed in lower case through the command line. See `env.yml` (which will be renamed `config.yml` as you requested).

File: scraper.py (hunk 6 above)
Comment (theyorubayesian) [reply scraper.py:208]: The `text` column is lengthy and may contain one or more commas already. Using `tsv` made it easier to understand what's in the file at a glance.

**Comment by theyorubayesian** [comment]
Thanks for the suggestions. I included them in the latest commit.

**Comment by keleog** [comment]
Looks good. 

Final change to be made: the `env.yml` file is still present. Also, we still use it in the code. 

Can we delete the `env.yml` file and change the variable in the code from `ENV` to `CONFIG`, while loading in `config.yml`.

File: env.yml (hunk 7 above)
Comment (keleog) [reply env.yml]: https://github.com/keleog/bbc_pidgin_scraper/pull/2#issuecomment-896125201

**Comment by theyorubayesian** [comment]
resolved these too.
